include/rear-panel/core/tasks/hardware_task.hpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Interface for the firmware-specific hardware tasks
+ */
+#pragma once
+
+#include "FreeRTOS.h"
+#include "common/core/debounce.hpp"
+#include "common/firmware/gpio.hpp"
+#include "rear-panel/core/messages.hpp"
+#include "rear-panel/core/queues.hpp"
+#include "rear-panel/core/tasks.hpp"
+#include "rear-panel/firmware/gpio_drive_hardware.hpp"
+
+namespace hardware_task {
+
+using TaskMessage = rearpanel::messages::HardwareTaskMessage;
+/*
+ * The task entry point.
+ */
+
+struct GpioInputState {
+    bool estop_in = false;
+    bool estop_aux1_det = false;
+    bool estop_aux2_det = false;
+    // TODO:
+    // door switch
+    // Aux device present
```

Comment:
separate from estop present? or is this out of date now

Reply:
Aux device is a separate pin that will be used by Hepa etc.